ui/types/vault/models/secret-engine/config.d.ts

```diff
@@ -6,22 +6,43 @@
 import type Model from '@ember-data/model';
 import type { ModelValidations } from 'vault/app-types';
 
```

Comment:
This is the "new" component test. A couple of things:
- I did my best to run for loops for the tests that made sense to run on each wif engine. 
- I did my best to test once if that tests wasn't dependent on engine type. 
- And I did my best to cover all engine specific things. 

All that to say, I did my best, _but_ my brain is fried after running through this so many times. If you catch an efficiency, please point it out.